nevergrad/optimization/coop.py

```diff
@@ -0,0 +1,136 @@
+        if self._config.groups is None:
+            self.species = list(self.parametrization._content.values())
+
+        self.species_iterator = cycle(range(len(self.species)))
+        self._uid_queue = base.utils.UidQueue()
+        self.populations: tp.List[tp.List[p.Parameter]] = [
+            [] for _ in range(len(self.species))
+        ]
+        self.candidates : tp.Dict[str, p.Parameter]
+        self.representatives: tp.List[p.Parameter] = []
+
+    def _internal_ask_candidate(self) -> p.Parameter:
+        idx = next(self.species_iterator)
+        current_species = self.species[idx]
+        current_population = self.populations[idx]
+
+        # Generate a representative of each species
+        if not self.representatives:
+            for sp, pop in zip(self.species, self.populations):
+                init_values = self._rng.normal(0, 1, sp.dimension)
+                rep = sp.spawn_child().set_standardized_data(init_values)
+                rep.heritage["lineage"] = rep.uid  # new lineage
+                self.representatives.append(rep)
+                pop[rep.uid] = rep
+
+            # TODO: Make this line work for Instrumentation
+            candidate_value = tuple(v.value for v in self.representatives)
+
+        # Random initialization
+        elif len(current_population) < self.llambda:  # initialization phase
+            init_values = self._rng.normal(0, 1, current_species.dimension)
+            candidate = current_species.spawn_child().set_standardized_data(init_values)
+
+            candidate.heritage["lineage"] = candidate.uid  # new lineage
+            current_population.append(candidate)
+            self.candidates[candidate.uid] = candidate
+            self._uid_queue.asked.add(candidate.uid)
+
+            # TODO: Make this work with Instrumentation
+            candidate_value = tuple(v.value for v in self.representatives[:idx]) \
+                              + (candidate.value,) \
+                              + tuple(v.value for v in self.representatives[idx+1:])
+
+        # Crossover or mutate
+        else:
+            pass
+
+
+        return self.parametrization.spawn_child(new_value=candidate_value)
+
+    def _internal_tell_candidate(self, candidate: p.Parameter, value: float) -> None:
+        pass
+
+    def _internal_tell_not_asked(self, candidate: p.Parameter, value: float) -> None:
+        pass
+
+
+class CooperativeOptimization(base.ConfiguredOptimizer, ):
+    def __init__(
+        self,
+        *,
+        groups: tp.Union[str, tp.List[tp.Tuple[tp.Union[int, str]]], None] = None,
+        crossover: str = "sbx",
+        mutation: str = "polynomial",
+        popsize: tp.Union[str, int] = "standard"
+    ):
+        super().__init__(_CooperativeOptimization, locals(), as_config=True)
+        self.groups = groups
+        self.crossover = crossover
+        self.mutation = mutation
+        self.popsize = popsize
+
+Coop = CooperativeOptimization().set_name("Coop")
```

Comment:
maybe << set_name("Coop", register=True) >> so that it's registered (i.e. users who run a loop over the registry of optimization methods see it).